R/hcai-impute.R

```diff
@@ -0,0 +1,133 @@
+#' @title
+#' Specify imputation methods for an existing recipe
+#'
+#' @description `hcai-impute` adds various imputation methods to an existing 
+#' recipe. Currently supports mean (numeric only), new_category (categorical 
+#' only), bagged trees, or knn. 
+#' @param rec_obj A recipe object. imputation will be added to the sequence of 
+#'  operations for this recipe.
+#' @param numeric_method Defaults to \code{"mean"}. Other choices are 
+#' \code{"bagimpute"} or \code{"knnimpute"}. 
+#' @param nominal_method Defaults to \code{"new_category"}. Other choices are 
+#' \code{"bagimpute"} or \code{"knnimpute"}. 
+#' @param numeric_params A named list with parmeters to use with chosen
+#' imputation method on numeric data. Options are \code{bag_model}, 
+#' \code{bag_options}, \code{knn_k}, \code{impute_with}, or \code{seed_val}. 
+#' See \link{step_bagimpute} or \link{step_knnimpute} for details.
+#' @param nominal_params A named list with parmeters to use with chosen
+#' imputation method on nominal data. Options are \code{bag_model}, 
+#' \code{bag_options}, \code{knn_k}, \code{impute_with}, or \code{seed_val}. 
+#' See \link{step_bagimpute} or \link{step_knnimpute} for details.
+#' @return An updated version of `recipe` with the new step
+#'  added to the sequence of existing steps.
+#'
+#' @export
+#' @import recipes
+
+#' @examples
+#' library(healthcareai)
+#' library(recipes)
+#' 
+#' n = 100
+#' d <- tibble::tibble(encounter_id = 1:n,
+#'             patient_id = sample(1:20, size = n, replace = TRUE),
+#'             hemoglobin_count = rnorm(n, mean = 15, sd = 1),
+#'             hemoglobin_category = sample(c("Low", "Normal", "High", NA), 
+#'                                          size = n, replace = TRUE),
+#'             disease = ifelse(hemoglobin_count < 15, "Yes", "No")
+#' )
+#' 
+#' # Initialize
+#' my_recipe <- recipe(disease ~ ., data = d)
+#' 
+#' # Create recipe
+#' my_recipe <- my_recipe %>%
+#'   hcai_impute(numeric_method = "mean",
+#'     nominal_method = "new_category")
+#' my_recipe
+#' 
+#' # Train recipe
+#' trained_recipe <- prep(my_recipe, training = d)
+#' 
+#' # Apply recipe
+#' data_modified <- bake(trained_recipe, newdata = d)
+#' countMissingData(data_modified)
+#' 
+hcai_impute <- function(rec_obj,
+                        numeric_method = "mean",
+                        nominal_method = "new_category",
+                        numeric_params = NULL,
+                        nominal_params = NULL) {
+  # Check to make sure rec_obj is the right type
+  if (class(rec_obj) != "recipe") {
```

Comment:
Weird that there isn't `is.recipe`. `inherits(rec_obj, "recipe")` would future proof a bit in case folks extend recipes.

Reply:
Changed.